### ISSUE TITLE
DB hook fires multiple times for create edge in TX mode

### ISSUE THREAD

**Issue by ylopotun** [orientechnologies/orientdb#4204]
I noticed that in TX mode DB hook's firing behaviour is inconsistent for base graph operations:
1) In case of create vertex it fires only once;
2) In case of set vertex property it fires only once;
3) In case of create edge it fires 3 times before commit (1 for edge itself, 1 for out vertex and 1 for in vertex) and 4 more times after commit (2 for edge itself, 1 for out vertex and 1 for in vertex).

Is it an expected behaviour? If so, why is it inconsistent throughout the operations?

**Comment by lvca** [comment]
This is normal, because when you create a regular edge (non lightweight), both vertices could be updated. In Transaction now should call the hook 3 times instead of 4 with the fix of yesterday. Could you try it?

**Comment by ylopotun** [comment]
But why when we create a vertex and then commit - the hook fires only 1 time? Also when we set a vertex property and commit - the hook fires only 1 time. It's only in case of edge creation that the hook fires again after commit...

**Comment by lvca** [comment]
After the commit? Ops, this is not expected. Are you sure?

**Comment by ylopotun** [comment]
Yes, sent a unit test via email (no option to attach it here).

**Comment by tglman** [comment]
actually running this test i saw that this is a duplicate of this: https://github.com/orientechnologies/orientdb/issues/3833

so this depend on this: https://github.com/orientechnologies/orientdb/issues/2620

i will look for a work around to not let this happen while commit, for the 2.1 release.

**Comment by tglman** [comment]
i tried to set up, a work around for avoid the multiple call, but the only way to fix it without regression is implement this: https://github.com/orientechnologies/orientdb/issues/2620, so i've to postpone this to 2.2

**Comment by prjhub** [comment]
The issue has been closed because no response has been received after a while. In case the issue is still valid, please reopen it or write a comment that explains why it's still valid and a way to reproduce it. Thank you.

**Comment by tglman** [comment]
this should be solved already in the develop branch, leaving it open because is missed the test case

**Comment by tglman** [comment]
implemented with test case closing

**Comment by lvca** [comment]
Reverted the commit

**Comment by tglman** [comment]
fixed closing
